Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -1604,6 +1605,46 @@ public void execution_deleteContext(@NotNull final String contextId) {
     }
   }
 
+  public RuntimeCompletionResult execution_getSuggestions(@NotNull final VirtualFile file,
```

Comment:
I guess we can get rid of `file` argument. AFAIK dummy files always have normalized line breaks, so no need to convert offset.

Reply:
Ah, OK.
I will remove it.
And for `logError` is is better to use `contextFile`.